src/CoreBundle/DataModel/TableConfigurationDataModel.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace ChameleonSystem\CoreBundle\DataModel;
+
+class TableConfigurationDataModel
+{
+    public function __construct(
+        readonly private string $id,
+        readonly private string $name,
+        readonly private string $cmsUsergroupId
+    ) {
+    }
```

Comment:
I am pretty sure you want to change that. I won't fight over it but the getters do not make much sense. I know, countless years of having to use them for this purpose makes it feel wrong to avoid them now, but they really are not needed anymore. :)

Reply:
I am not so sure. Currently, I am getting a nice code hint when using the getter. I have no idea how to ensure that I still get that when accessing the public property.

Reply:
It should work for everything except typed arrays - so basically every type annotation that is supported by php.

If you need the annotation for types in arrays you could annotate them like this:

```php
final class TestClass {

    public function __construct(
        /** @var array<string, string> */
        public readonly array $arr
    ){}

}
```

Would this work for you?

Reply:
yes - and no. I did change `\ChameleonSystem\CoreBundle\DataModel\TableConfigurationDataModel` to use public readonly. But the CmsUserModel implements `UserInterface, PasswordAuthenticatedUserInterface` - which dictate that it has methods such as getRoles, getPassword, getSalt and getUsername. I find it very strange to use getters for some properties but not for others - so in this case I am keeping it

Reply:
that's a good point.